package-examples/kustomize-pkg/README.md

```diff
@@ -0,0 +1,163 @@
+# kustomize-pkg
+
+This is a simple package that shows how kpt packages can be used instead of remote bases in kustomize.  That allows you to take advantage of kpt's rebase and local, in-place edits.
+
+## Steps
+
+1. [Fetch the package](#fetch-the-package)
+2. [View the package contents](#view-the-package-contents)
+3. [kustomize the config](#kustomize-the-config)
+4. [Render the kustomization](#render-the-kustomization)
+5. [Apply the package](#apply-the-package)
+
+### Fetch the package
+
+Get the example package on to local using `kpt pkg get`. Note that this package is for this example, wihin that package we are also using the nginx sub-package which is an alternative to having a remote base. 
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/kustomize-pkg
+
+fetching package /package-examples/kustomize-pkg from https://github.com/GoogleContainerTools/kpt to kustomize-pkg
+```
+
+
+### View the package contents
+
+List the package contents in a tree structure.
+
+```shell
+$ kpt pkg tree kustomize-pkg/
+
+PKG: kustomize-pkg
+├── [Kptfile]  Kptfile kustomize-pkg
+├── PKG: nginx
+│   ├── [Kptfile]  Kptfile nginx
+│   ├── [deployment.yaml]  Deployment my-nginx
+│   └── [svc.yaml]  Service my-nginx-svc
+├── dev
+│   ├── [kustomization.yaml]  Kustomization 
+│   └── [pass-patch.yaml]  Deployment deployment-patch
+└── prod
+    ├── [kustomization.yaml]  Kustomization 
+    └── [pass-patch.yaml]  Deployment deployment-patch
+```
+
+### kustomize the config
+
+We recommend that you keep the kustomize instructions to rendering only such as adding a namespace, transforming or applying a patch.  The second you mix kpt packages and remote bases you will be missing out on a big advantage of having a guaranteed stable base.
```

Comment:
Can we make this clearer? It is not clear to me how using kustomize remote bases would necessarily be bad here, other than challenges around mixing packages fetched with kpt and remote bases with kustomize.

Reply:
That gets convoluted though as you're effectively using two package managers concurrently. It's hard even for me to reason about it. Instead, you want to have a clean conceptual separation of concerns. The price is that you cannot just drop any arbitrary existing kustomization resource since it may use remote bases.

+1 to making this intentional limiting of kustomize features more precise. Explicitly state what features of kustomize is not recommended?

Reply:
I added stronger language, but I think enumerating every feature of kustomize is going to create a wackamole game.  There are new features added and deprecated as we speak and it's best to have a larger umbrella like hydration and packaging in order to avoid updating the sample language every time something changes in kustomize.